docs/readme-sync/v4.0/deploy-as-a-microservice/030 - use-optimizely-agent/index.md

```diff
@@ -0,0 +1,112 @@
+---
+title: "Use Optimizely Agent"
+excerpt: ""
+slug: "use-optimizely-agent"
+hidden: false
+metadata: 
+  title: "How to use Optimizely Agent - Optimizely Full Stack"
+createdAt: "2020-02-21T17:44:28.054Z"
+updatedAt: "2020-04-08T21:26:30.308Z"
+---
+
+Optimizely Agent provides [APIs](https://library.optimizely.com/docs/api/agent/v1/index.html) that enable running feature flag rules, such as A/B tests and targeted flag deliveries. Agent provides equivalent functionality to all our SDKs. At its core is the [Optimizely Go SDK](doc:go-sdk). 
+
+### Running feature flag rules
+
+
+The Decide [endpoint](https://library.optimizely.com/docs/api/agent/v1/index.html#operation/decide) buckets a user into a feature flag variation (choosing between multiple enabled or one disabled variation) as part of a flag rule. Flag rules include A/B tests and targeted feature flag deliveries. To run a flag rule, use:
+
+`POST /v1/decide?keys={flagKey}`
+
+In the request `application/json` body, include the `userID` and any `decideOptions`. The full request looks like this:
+
+```curl
+curl --request POST 'http://localhost:8080/v1/decide' \
+--header 'Content-Type: application/json' \
+--header 'X-Optimizely-SDK-Key: <YOUR_SDK_KEY>' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+"userId": "test-user"
+"decideOptions": [
+   "INCLUDE_REASONS"
+]
+}'
+```
+
+TODO: please review above and below request/response examples, I didn't test them, just looked at the Pull request for decide and made some guesses!! -FE
```

Comment:
looks fine to me, here is one i tested with postman:

curl --location --request POST 'http://localhost:8080/v1/decide?keys=feature_enabled&keys=feature_disabled' \
--header 'X-Optimizely-SDK-Key: NCMRF8PXTMNCvGNUNKm32' \
--header 'Accept: text/event-stream' \
--header 'Content-Type: application/json' \
--data-raw '{
    "decideOptions": [
  ],
  "userId": "string",
  "userAttributes": {
    "additionalProp1": {}
  }
}'

Reply:
thanks, I used that and just subbed out the flag names/sdk key
```curl
curl --location --request POST 'http://localhost:8080/v1/decide?keys=YOUR_FLAG_1&keys=YOUR_FLAG_2'
--header 'X-Optimizely-SDK-Key: YOUR_SDK_KEY'
--header 'Accept: text/event-stream'
--header 'Content-Type: application/json'
--data-raw '{
"decideOptions": [
],
"userId": "string",
"userAttributes": {
"additionalProp1": {}
}
}'
```